libs/core/state/src/operation/selectors.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  createSelector,
+  MemoizedSelector,
+  Selector,
+} from '@ngrx/store';
+
+import { DaffState } from '../states/public_api';
+import { DaffOperationState } from './state';
+
+/**
+ * Selectors for an operation state.
+ */
+export interface DaffOperationStateSelectors<
+  TRootState,
+  TState extends DaffOperationState = DaffOperationState
+> {
+  /**
+   * Selects the loading state enum.
+   */
+  selectLoadingState: MemoizedSelector<TRootState, TState['loading']>;
+  /**
+   * Selects whether the operation state is in any of the loading states.
+   */
+  selectLoading: MemoizedSelector<TRootState, boolean>;
+  /**
+   * Selects whether the operation state is resolving.
+   */
+  selectResolving: MemoizedSelector<TRootState, boolean>;
+  /**
+   * Selects whether the operation state is mutating.
+   */
+  selectMutating: MemoizedSelector<TRootState, boolean>;
+  /**
+   * Selects the errors in the operation state.
+   */
+  selectErrors: MemoizedSelector<TRootState, TState['errors']>;
+  /**
+   * Selects whether the operation state has any errors.
+   * If so, it should be considered to be in an "error" state.
+   */
+  selectHasErrors: MemoizedSelector<TRootState, boolean>;
+}
+
+/**
+ * Creates a set of selectors for an operation state.
+ *
+ * @param selectState The feature selector for the operation state.
+ */
+export const daffOperationStateSelectorFactory = <
```

Comment:
This is a dev time function only, it only provides organizational use to us. Is there a way to leverage the compiler's code folding feature to remove this function before runtime?

Reply:
the selector factory is the entrypoint into the selector portion of this feature. Are you suggesting that I memoize this factory and only export the memoized version?

Reply:
changed to named function